--- a/include/havoqgt/breadth_first_search.hpp
+++ b/include/havoqgt/breadth_first_search.hpp
@@ -11,6 +11,8 @@
 #include <havoqgt/detail/visitor_priority_queue.hpp>
 #include <boost/container/deque.hpp>
 
+#include <limits>
+
 namespace havoqgt { 
 
 template <typename Visitor>
--- a/src/run_page_rank.cpp
+++ b/src/run_page_rank.cpp
@@ -20,6 +20,7 @@
 #include <utility>
 #include <algorithm>
 #include <functional>
+#include <limits>
 
 #include <boost/interprocess/managed_heap_memory.hpp>
 
--- a/include/havoqgt/page_rank.hpp
+++ b/include/havoqgt/page_rank.hpp
@@ -12,6 +12,7 @@
 #include <havoqgt/visitor_queue.hpp>
 #include <boost/container/deque.hpp>
 #include <vector>
+#include <limits>
 
 namespace havoqgt {
 
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -18,7 +18,7 @@ function ( add_mpi_ctest_no_gtest test_name )
   SET( num_procs 2)
   WHILE( 8 GREATER ${num_procs} )
     add_test( "${test_name}_${num_procs}" ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG}
-            ${num_procs} ${MPIEXEC_PREFLAGS} "${CMAKE_CURRENT_BINARY_DIR}/${test_exe}")
+            ${num_procs} "--oversubscribe" ${MPIEXEC_PREFLAGS} "${CMAKE_CURRENT_BINARY_DIR}/${test_exe}")
     MATH( EXPR num_procs "${num_procs} * 2" ) #double num_procs
   ENDWHILE()
 endfunction()
@@ -28,4 +28,4 @@ add_mpi_ctest_no_gtest( bfs_delegate )
 add_mpi_ctest_no_gtest( cc )
 add_mpi_ctest_no_gtest( cc_delegate )
 
-add_subdirectory(datasets)
\ No newline at end of file
+add_subdirectory(datasets)
